test/e2e/lib/components/abstract-editor-component.js

```diff
@@ -6,6 +6,26 @@ import * as driverHelper from '../driver-helper';
  * Abstract class containing shared code for editor-related classes.
  */
 export default class AbstractEditorComponent extends AsyncBaseContainer {
+	static async Expect( driver, editorType ) {
+		const page = new this( driver, null, editorType );
+		await page._expectInit();
+
+		// The `wpcom-editor-welcome-tour-frame` popup could get in the way
+		// of the actual publish click, so we need to check if it's visible
+		// and click `Skip` to close it.
+		await page.dismissNuxWelcomeModalIfDisplayed();
+
+		return page;
+	}
+
+	async _preInit() {
+		if ( this.editorType !== 'iframe' ) {
+			return;
+		}
+		await this.driver.switchTo().defaultContent();
+		await driverHelper.waitUntilAbleToSwitchToFrame( this.driver, this.editoriFrameLocator );
```

Comment:
I think we are having an issue because `this.editorType` and `this.editorIframeLocator` are not defined in this class / abstract component.  Previously when this method was in the site-editor-component, both of these were defined in the constructor.

Reply:
I thought this was the culprit for the recent test failures we discussed in slack, but it seems to fail intermittently with and without this change.  It does still seem to be an issue with accurately and reliably switching to the correct frame though.